src/tr1/game/savegame/savegame_legacy.c

```diff
@@ -665,8 +665,12 @@ bool Savegame_Legacy_LoadFromFile(MYFILE *fp, GAME_INFO *game_info)
     }
 
     M_ReadLara(&g_Lara);
-    M_Read(&g_FlipEffect, sizeof(int32_t));
-    M_Read(&g_FlipTimer, sizeof(int32_t));
+    int32_t flip_effect;
+    int32_t flip_timer;
+    M_Read(&flip_effect, sizeof(int32_t));
```

Comment:
Let's use M_ReadS32 as a safer alternative.

Reply:
We don't have this in TR1's legacy save module. Should we introduce it from TR2?

Reply:
Ideally, yes,  but since this would be about changing the entire file, let's do that separately.